pkg/api/validation/schema.go

```diff
@@ -102,11 +109,18 @@ func (s *SwaggerSchema) ValidateObject(obj interface{}, apiVersion, fieldName, t
 	}
 	fields, ok := obj.(map[string]interface{})
 	if !ok {
-		return fmt.Errorf("expected object of type map[string]interface{} as value of %s field", fieldName)
+		return append(allErrs, fmt.Errorf("expected object of type map[string]interface{} as value of %s field", fieldName))
 	}
 	if len(fieldName) > 0 {
 		fieldName = fieldName + "."
 	}
+	//handle required fields
+	for _, requiredKey := range model.Required {
+		_, ok := fields[requiredKey]
```

Comment:
Also, this change also makes the code more consistent with code elsewhere in Kubernetes.